modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/form.js

```diff
@@ -239,19 +239,24 @@ AUI.add(
 				_valueEditingLanguageId: function() {
 					var instance = this;
 
-					var portletNamespace = instance.get('portletNamespace');
+					var languageId;
 
-					var languageId = instance._getURLParameter(
-						portletNamespace,
-						'languageId'
-					);
+					var formNode = instance.getFormNode();
 
-					if (!languageId) {
-						languageId = themeDisplay.getLanguageId();
+					if (formNode) {
+						var portletNamespace = instance.get('portletNamespace');
 
-						if (!languageId) {
-							languageId = instance.get('defaultLanguageId');
-						}
+						var languageIdNodeId = portletNamespace + 'languageId';
+
+						formNode.get('children').each(function(node) {
```

Comment:
In my fix suggestion, I've used the method `each`, however would be better use a method that iterates only over the necessary elements, like [some](https://yuilibrary.com/yui/docs/api/classes/NodeList.html#method_some). You can also check an example of how to use this method [here](https://github.com/liferay/liferay-portal/blob/f9d4a81205d64b025ef9718c17760e2d03aeed6d/modules/apps/calendar/calendar-web/src/main/resources/META-INF/resources/js/simple_menu.js#L42-L50). 😉

Reply:
Great, I didn't know about this method to be honest! The reason of my delayed response is because I've noticed that there is a change here in the DDM Form Renderer on the frontend side, caused by LPS-93027 & LPS-97127. This issue is not reproduceable anymore on master because of the two LPPs mentioned earlier so I believe you can just close this PR.

However we're still having the old setup on 7.1.x so I'll update my modification with your suggestion there and will send you a new PR for 7.1.x.

Thanks!
Robert

Reply:
Great, Robert! Thanks. 🙂